lib/statesman/config.rb

```diff
@@ -5,14 +5,26 @@
 
 module Statesman
   class Config
-    attr_reader :adapter_class
+    attr_reader :adapter_class, :mysql_gaplock_protection
 
     def initialize(block = nil)
+      @mysql_gaplock_protection = db_mysql?
       instance_eval(&block) unless block.nil?
     end
 
     def storage_adapter(adapter_class)
       @adapter_class = adapter_class
```

Comment:
I _think_ we should probably set the mysql_gaplock_protection variable, if it's not nil, when we assign the adapter. I'm not sure it's fully safe for us to tap ActiveRecord::Base for this when we allow people to provide specific adapter classes, or have I misunderstood?

Reply:
Not sure I fully understand what you're suggesting. Are you suggesting we defer `mysql?` logic to the adapter classes? Or just removing that logic from the initialiser. I agree that having AR specific stuff in the config isn't great.